aws/tfstate-backend/scripts/destroy.sh

```diff
@@ -0,0 +1,30 @@
+# Start with a clean slate
+rm -rf .terraform terraform.tfstate
+
+# Init terraform with S3 state enabled. Assumes state was previously initialized.
+init-terraform
+
+# Unmount remote bucket (if mounted)
+s3 unmount
+
+# Store the current state so we can destroy resources without catch-22
+terraform state pull > terraform.tfstate
```

Comment:
do we first unmount the bucket with `s3 unmount` above and then try to pull the remote state from the unmounted bucket?

Reply:
Mounting the S3 bucket has no bearing on how state is managed. The bucket is only mounted as a "nice to have" for occasionally writing things like SSH keys somewhere.